Alcatraz/Controllers/ATZPluginWindowController.m

```diff
@@ -89,10 +89,18 @@ - (BOOL)userNotificationCenter:(NSUserNotificationCenter *)center shouldPresentN
 - (IBAction)installPressed:(ATZFillableButton *)button {
     ATZPackage *package = [self.tableViewDelegate tableView:self.tableView objectValueForTableColumn:0 row:[self.tableView rowForView:button]];
     
-    if (package.isInstalled)
-        [self removePackage:package andUpdateControl:button];
-    else
+    if (package.isInstalled) {
+        if ([package isKindOfClass:[ATZPlugin class]] && [(ATZPlugin *)package isBlacklisted]) {
```

Comment:
This breaks the idea of polymorphism. We shouldn't be looking up specific package classes